### PR TITLE
[ACTIONS] New Qdrant actions

### LINKED ISSUE #12122: [Components] qdrant
# qdrant
## URLs
- https://qdrant.github.io/qdrant/redoc/index.html
#### Prompt
Removes specific points from a collection. Required props: collection ID, points IDs.
#### URLs
- https://qdrant.github.io/qdrant/redoc/v1.9.x/openapi.json
- https://qdrant.github.io/qdrant/redoc/index.html?v=master
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.9.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.8.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.7.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.6.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.5.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.4.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.3.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.2.x
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.1.3
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.1.2
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.1.1
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.1.0
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.0.3
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.0.2
- https://qdrant.github.io/qdrant/redoc/index.html?v=v1.0.1
#### Prompt
Adds a single point to a specific collection. Required props: collection ID, point data. Optional props: point ID.
#### URLs
- https://qdrant.github.io/qdrant/redoc/v1.9.x/openapi.json
#### Prompt
Finds the most similar vectors within a collection. Required props: collection ID, vectors. Optional props: search parameters.
#### URLs
- https://qdrant.github.io/qdrant/redoc/index.html
- https://qdrant.github.io/qdrant/redoc/index.html?v=master
- https://qdrant.github.io/qdrant/redoc/v1.9.x/openapi.json

### PR DESCRIPTION
**Pull request by Anush008**
## Description

This PR adds actions to add, search, retrieve, and delete points from a Qdrant collection. Along with a generic Qdrant API call action for extensibility.

Resolves #12122

Related to https://github.com/PipedreamHQ/pipedream/issues/12096.

## Summary by CodeRabbit

- **New Features**
  - Introduced actions for Qdrant:
    - Delete Points
    - Get Points
    - Search Points
    - Upsert Point
  - Added utility functions for handling Qdrant point IDs and vectors.

- **Documentation**
  - Added README for Qdrant component with an overview of its features.

- **Chores**
  - Updated `package.json` to version 0.1.0 and added `@qdrant/js-client-rest` dependency.

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/qdrant/qdrant.app.mjs [hunk 1]
```diff
@@ -1,11 +1,70 @@
...
+        url: this.$auth.url,
+        apiKey: this.$auth.api_key,
+      };
+    },
+    async listCollections() {
+      const client = new QdrantClient(this.getCredentials());
+
+      const collections = (await client.getCollections()).collections;
+
+      return collections.map((collection) => collection.name);
```
[the remark this answers is not in the document]
Comment (Anush008) [reply components/qdrant/qdrant.app.mjs:67]: I did have something similar.

But when using this in different files, you miss the type hints of `QdrantClient`. 
So I'm instantiating in-line.

File: components/qdrant/package.json
[the remark this answers is not in the document]
Comment (Anush008) [reply components/qdrant/package.json:1]: https://github.com/PipedreamHQ/pipedream/pull/12135/commits/1aac2abe0acd1ed7cb061def69e83433b23e709b

**Comment by Anush008** [comment]
Hi @jcortes. May I know more about the expected changes?

**Comment by michelle0927** [comment]
> Hi @jcortes. May I know more about the expected changes?

@Anush008 Thanks for your contribution! Here is the link to the QA report with the requested changes: https://vunguyenhung.notion.site/Components-qdrant-4a2e618f59a54af285ab8e72c2af8143

**Comment by Anush008** [comment]
Hi @michelle0927. Thanks for the page. It was very comprehensive and helpful.

The failing delete with filters issue should be resolved now.

**Pull request merged**